src/build/app/mod.rs

```diff
@@ -1481,40 +1470,160 @@ impl<'b> App<'b> {
+
+            if let Some(reqs) = &arg.r_ifs {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == req.0)
+                            || self.groups.iter().any(|x| x.id == req.0),
+                        "Argument or group specified in 'required_if*' for '{}' does not exist",
+                        arg.name,
+                    );
+                }
+            }
+
+            if let Some(reqs) = &arg.r_unless {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == *req)
+                            || self.groups.iter().any(|x| x.id == *req),
+                        "Argument or group specified in 'required_unless*' for '{}' does not exist",
+                        arg.name,
+                    );
+                }
+            }
+
+            // blacklist
+            if let Some(reqs) = &arg.blacklist {
+                for req in reqs {
+                    assert!(
+                        self.args.args.iter().any(|x| x.id == *req)
+                            || self.groups.iter().any(|x| x.id == *req),
+                        "Argument or group specified in 'conflicts_with*' for '{}' does not exist",
+                        arg.name,
+                    );
+                }
+            }
+
+            if arg.is_set(ArgSettings::Last) {
+                assert!(
+                    arg.long.is_none(),
+                    "Flags or Options may not have last(true) set. '{}' has both a long and last(true) set.",
+                    arg.name
+                );
+                assert!(
+                    arg.short.is_none(),
+                    "Flags or Options may not have last(true) set. '{}' has both a short and last(true) set.",
+                    arg.name
+                );
+            }
+
+            assert!(
+                !(arg.is_set(ArgSettings::Required) && arg.global),
+                "Global arguments cannot be required.\n\n\t'{}' is marked as global and required",
+                arg.name
+            );
+        }
+
+        for group in &self.groups {
+            // Name conflicts
+            assert!(
+                self.groups.iter().filter(|x| x.id == group.id).count() < 2,
+                "Argument group name must be unique\n\n\t'{}' is already in use",
+                group.name,
+            );
+
+            // Groups should not have naming conflicts with Args
+            assert!(
+                !self.args.args.iter().any(|x| x.id == group.id),
+                "Argument group name '{}' must not conflict with argument name",
+                group.name,
+            );
+
+            // Args listed inside groups should exist
+            for arg in &group.args {
+                assert!(
+                    self.args.args.iter().any(|x| x.id == *arg),
+                    "Argument group '{}' contains non-existent argument",
```

Comment:
Without knowing which argument does not exist, it can be hard to figure out at times.

Reply:
Can't do it because `pub(crate) args: Vec<Id>,`, which means the argument name is `Id` and not string.

Reply:
I have an idea: Id can be represented as
```rust
struct Id {
    #[cfg(debug_assertions)]
    name: String,
    id: u64
}
```

instead of `type Id = u64`, possibly along with `Deref<u64>`, `PartialEq` and so on. This would also allow for nicer `Debug` impls for `Arg & App`.

I'll work on it after I'm done with interpolation.